packages/generator/README.md

```diff
@@ -21,3 +27,130 @@ For installation and usage, see [Measurement Plug-In SDK for Python (`ni-measure
+        
+        > **Note**: For multiple measurement client generation, `--module-name` and `--class-name` are ignored and derived from the service class of each measurement. So, ensure that the measurement service class name adheres to proper naming conventions.
+
+    2. Run this command to create measurement clients for all registered measurements.
+
+        `ni-measurement-plugin-client-generator --all`
+
+        > **Note**: `--directory-out` can be provided for this command.
+
+    3. Run this command to create measurement clients for any registered measurements interactively.
+
+        `ni-measurement-plugin-client-generator --interactive`
+
+3. The generated client includes four APIs: `measure`, `stream_measure`, `register_pin_map`, and `cancel`. The usage of these APIs is discussed in the section ["Steps to Run/Debug the Measurement Client".](#steps-to-rundebug-the-measurement-client)
+
+> **Note**:
+> - The Measurement Plug-In Client is compatible with all datatypes supported by the Measurement Plug-In.
+> - The Double XY datatype is not supported for measurement configurations (inputs).
+> - For Enum datatypes, the generated enum class names will be the measurement parameter name suffixed with 'Enum'. For instance, if the measurement parameter name is 'Enum In', the generated enum in the client will be `EnumInEnum'.
+> - Ring control in LabVIEW measurements will be represented as numeric datatypes in the generated client.
+
+---
+
+## Steps to Run/Debug the Measurement Client
+
+1. Make sure the required measurement service is running before interacting with it via the client.
+
+2. Use the client APIs from the ["Developing a Minimal Python MeasurementClient"](#developing-a-minimal-python-measurement-client) section.
+
+    1. For non-streaming measurements, use `measure` method.
+
+        ``` python
+        from sample_measurement_client import SampleMeasurementClient
+        
+        client = SampleMeasurementClient()
+        outputs = client.measure()
+        ```
+
+    2. For streaming measurements, use `stream_measure` method.
+
+        ``` python
+        from sample_measurement_client import SampleMeasurementClient
+        
+        client = SampleMeasurementClient()
+        outputs_itr = client.stream_measure()
+        for index, outputs in enumerate(outputs_itr):
+            print(f"outputs[{index}] = {outputs}")
+        ```
+
+    3. If a measurement requires a pin map, it can be registered using the `register_pin_map` method. By default, `sites` is set to 0.
+
+        ``` python
+        from sample_measurement_client import SampleMeasurementClient
+        
+        client = SampleMeasurementClient()
+        client.register_pin_map(pin_map_path)
+        output = client.measure()
+        ```
+        - Alternatively, you can provide a pin map context through the `pin_map_context` property. Similarly, sites can be provided through the `sites` property.
```

Comment:
Why: when calling a measurement service from another measurement, you can pass the first measurement's pin map context to the second measurement.